src/NuGet.Core/NuGet.Protocol/Plugins/PluginException.cs

```diff
@@ -29,5 +30,9 @@ public PluginException(string message, Exception innerException)
             : base(message, innerException)
         {
         }
+
+        private PluginException(SerializationInfo info, StreamingContext context) : base(info, context)
```

Comment:
This constructor is marked private because the type is sealed. Guidance - https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2229 says `The type is sealed and the access modifier for its serialization constructor is not private. ` This means I think the serialization constructor should be `private` for `sealed` types.

I verified in a console application to ensure serialization and deserialization works properly even though constructor is private.